claripy/backends/backend_z3.py

```diff
@@ -917,7 +930,7 @@ def _max(self, expr, extra_constraints=(), solver=None, model_callback=None):
             # TODO: is this assumption correct?
             # here it's not safe to call directly the z3 low level API since it might happen that the argument is an
             # integer and not a BV
-            solver.add(z3.UGT(expr, middle), z3.ULE(expr, hi))
```

Comment:
One strange thing I noticed while working on this: it's weird how max uses GT and LE while min uses GE and LE. Is there a reason for this?